datasets/new_york/pipelines/_images/run_csv_transform_kub_nypd_mv_collisions/csv_transform.py

```diff
@@ -0,0 +1,317 @@
+        df[field_name] = df[field_name].apply(
+            lambda x: convert_dt_format(str(x), from_format, to_format)
+        )
+    return df
+
+
+def convert_dt_format(
+    dt_str: str, from_format: str, to_format: str = "%Y-%m-%d %H:%M:%S"
+) -> str:
+    if not dt_str or str(dt_str).lower() == "nan" or str(dt_str).lower() == "nat":
+        dt_str = ""
+        return dt_str
+    else:
+        if from_format == "%Y%m%d":
+            year = dt_str[0:4]
+            month = dt_str[4:6]
+            day = dt_str[6:8]
+            dt_str = f"{year}-{month}-{day} 00:00:00"
+            from_format = "%Y-%m-%d %H:%M:%S"
+        elif len(dt_str.strip().split(" ")[1]) == 8:
+            # if format of time portion is 00:00:00 then use 00:00 format
+            dt_str = dt_str[:-3]
+        elif (len(dt_str.strip().split("-")[0]) == 4) and (
+            len(from_format.strip().split("/")[0]) == 2
+        ):
+            # if the format of the date portion of the data is in YYYY-MM-DD format
+            # and from_format is in MM-DD-YYYY then resolve this by modifying the from_format
+            # to use the YYYY-MM-DD.  This resolves mixed date formats in files
+            from_format = "%Y-%m-%d " + from_format.strip().split(" ")[1]
+        return datetime.datetime.strptime(dt_str, from_format).strftime(to_format)
+
+
+def add_crash_timestamp(
+    df: pd.DataFrame, new_crash_field: str, crash_date_field: str, crash_time_field: str
+) -> pd.DataFrame:
+    logging.info(
+        f"add_crash_timestamp '{new_crash_field}' '{crash_date_field}' '{crash_time_field}'"
+    )
+    df[new_crash_field] = df.apply(
+        lambda x, crash_date_field, crash_time_field: crash_timestamp(
+            x["" + crash_date_field], x["" + crash_time_field]
+        ),
+        args=[crash_date_field, crash_time_field],
+        axis=1,
+    )
+    return df
+
+
+def crash_timestamp(crash_date: str, crash_time: str) -> str:
+    # if crash time format is H:MM then convert to HH:MM:SS
+    if len(crash_time) == 4:
+        crash_time = f"0{crash_time}:00"
+    return f"{crash_date} {crash_time}"
+
+
+def save_to_new_file(df: pd.DataFrame, file_path: str) -> None:
+    df.to_csv(file_path, index=False)
+
+
+def append_batch_file(
+    batch_file_path: str,
+    target_file_path: str,
+    skip_header: bool,
+    truncate_file: bool
+) -> None:
+    with open(batch_file_path, "r") as data_file:
+        if truncate_file:
+            target_file = open(target_file_path, "w+").close()
+        with open(target_file_path, "a+") as target_file:
+            if skip_header:
+                logging.info(
+                    f"Appending batch file {batch_file_path} to {target_file_path} with skip header"
+                )
+                next(data_file)
+            else:
+                logging.info(f"Appending batch file {batch_file_path} to {target_file_path}")
+            target_file.write(data_file.read())
+            data_file.close()
```

Comment:
Not sure why you are manually closing these two files. The `with` clause automatically closes a file when it reaches the end of the code block of that `with`.

Reply:
Done